xml/System.Windows.Threading/Dispatcher.xml

```diff
@@ -1077,6 +1077,10 @@
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentNullException">
+          <paramref name="method" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="timeout" /> is a negative number other than -1 milliseconds, which represents infinite wait, and you're invoking across threads.</exception>
```

Comment:
This is the wrong place to add "which represents infinite wait". An infinite wait should be documented in the params section, not the exceptions. The params section should also note how negative numbers are interpreted in cross-thread and same-thread calls.

Reply:
I agree.  
What really happens is that -1 represents infinite wait.  In a same-thread call, any other negative timeout is converted to -1, resulting in infinite wait.  In a cross-thread call, any other negative timeout throws ArgumentOutOfRangeException.  (This is for compat with 4.0, which behaved that way by accident of implementation.)